packages/devtools_app/lib/src/service/editor/editor_client.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'api_classes.dart';
+
+/// An interface to services provided by an editor.
+///
+/// Changes made to the editor services/events should be considered carefully to
+/// ensure they are not breaking changes to already-shipped editors.
+abstract class EditorClient {
+  Future<void> close();
+  bool get supportsGetDevices;
+  bool get supportsSelectDevice;
+  bool get supportsHotReload;
+  bool get supportsHotRestart;
+  bool get supportsOpenDevToolsPage;
+  bool get supportsOpenDevToolsExternally;
+
+  /// A stream of [EditorEvent]s from the editor.
+  Stream<EditorEvent> get event;
+
+  /// Gets the set of currently available devices from the editor.
+  Future<List<EditorDevice>> getDevices();
+
+  /// Gets the set of currently active debug sessions from the editor.
+  Future<List<EditorDebugSession>> getDebugSessions();
+
+  /// Requests the editor selects a specific device. It should not be assumed
```

Comment:
nit: new line after first sentence of dart doc

Reply:
Done in 8ec9692b9